controllers/route-helpers.js

```diff
@@ -0,0 +1,85 @@
+const { compact, compose, concat, filter, flatMap, getOr, map, pick, sortBy, uniqBy } = require('lodash/fp');
+const contentApi = require('../services/content-api');
+const routes = require('./routes');
+
+const sortedUniqByPath = compose(sortBy('path'), uniqBy('path'));
+const isLive = route => route.live === true;
+
+/**
+ * Build a flat list of all canonical routes
+ * Combines application routes and routes defined by the CMS
+ */
+function getCanonicalRoutes({ includeDraft = false } = {}) {
+    const routerCanonicalUrls = flatMap(section => {
+        const withoutWildcards = filter(_ => _.path.indexOf('*') === -1);
+        const mapSummary = map((page, key) => {
+            return {
+                title: key,
+                path: section.path + page.path,
+                live: page.live
+            };
+        });
+
+        return compose(mapSummary, withoutWildcards)(section.pages);
+    })(routes.sections);
+
+    return contentApi.getRoutes().then(cmsCanonicalUrls => {
+        const combined = concat(routerCanonicalUrls, cmsCanonicalUrls);
+        const filtered = includeDraft === true ? combined : combined.filter(isLive);
```

Comment:
Could just do:
`const filtered = (includeDraft) ? combined : combined.filter(isLive);`

Reply:
Personal preference but I like the explicitness